modules/fedimint-mint-client/src/lib.rs

```diff
@@ -843,6 +851,70 @@ impl MintClientModule {
         stream.next_or_pending().await
     }
 
+    /// Provisional implementation of note consolidation
+    ///
+    /// When a certain denomination crosses the threshold of notes allowed,
+    /// spend some chunk of them as inputs.
+    pub async fn consolidate_notes(
+        &self,
+        dbtx: &mut DatabaseTransaction<'_>,
+        operation_id: OperationId,
+    ) -> anyhow::Result<Vec<ClientInput<MintInput, MintClientStateMachines>>> {
+        let notes_by_denomination: BTreeMap<Amount, Vec<SpendableNote>> = dbtx
+            .find_by_prefix_sorted_descending(&NoteKeyPrefix)
+            .await
+            .map(|(key, note)| (key.amount, note))
+            .fold(
+                BTreeMap::<Amount, Vec<SpendableNote>>::new(),
+                |mut acc, (amount, note)| async move {
+                    acc.entry(amount).or_default().push(note);
+                    acc
+                },
+            )
+            .await;
```

Comment:
This loads all the notes to memory, we tried to avoid that for selection, I don't remember if that was premature optimization or if we ran into a particular problem.